api/src/paths/project/{projectId}/survey/{surveyId}/devices/index.ts

```diff
@@ -137,6 +138,14 @@ export function createDevice(): RequestHandler {
 
       const telemetryDeviceService = new TelemetryDeviceService(connection);
 
+      // Check whether device already exists in Survey
+      const device = await telemetryDeviceService.findDeviceBySerial(surveyId, serial, device_make_id);
```

Comment:
Note: There is already a unique constraint on the device table for (survey_id, serial, device_make_id)

Reply:
I did this to catch the would-be error to give a useful error message. I figured the goal was more obvious than waiting for the database error, but let me know if you think we should do it differently.